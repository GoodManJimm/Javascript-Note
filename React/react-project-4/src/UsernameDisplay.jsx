export function UsernameDispay({username})
{
    return(
        <div>
            <h1>{username}</h1>
            <h1>{username}</h1>
        </div>
    )
}