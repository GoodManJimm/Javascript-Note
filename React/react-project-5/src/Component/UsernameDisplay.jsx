import { useEffect, useState } from "react"

export function UsernameDisplay({username})
{

    const [timePassed, setTimePassed] = useState(false)

    useEffect(() => {
        setTimeout(() => {
            setTimePassed(true)
        },1000)
    },[])
    return(
        <div>
            <span>{timePassed && username}</span>
        </div>
    )
}