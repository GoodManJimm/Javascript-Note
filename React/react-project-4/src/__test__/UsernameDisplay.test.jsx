import {describe,it, test, expect} from "vitest"
import {render, screen} from "@testing-library/react"
import { UsernameDispay } from "../UsernameDisplay"

describe('UsernameDisplay', () => {
    it('should render username', () => {
    //     const result = render(<UsernameDispay username="ha" />)
    //     // expect(result.container).toMatchSnapshot();
    // //    expect(screen.getByText("ha")).toBeInTheDocument();
    // //    expect(screen.getByText(/h/)).toBeInTheDocument();
       
    //    const elements = screen.getAllByText("ha");
    //    elements.forEach((element) => expect(element).toBeInTheDocument())
    })
})