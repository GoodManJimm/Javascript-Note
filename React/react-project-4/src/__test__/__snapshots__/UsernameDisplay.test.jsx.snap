// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UsernameDisplay > should render username 1`] = `
<div>
  <div>
    <h1>
      ha
    </h1>
  </div>
</div>
`;
